docs/android/tips/leak_canary.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "Leak Canary 🐤"
+sidebar_label: "Leak Canary"
+---
+
+## How to disable Leak Canary in debug builds
+
+[LeakCanary](https://square.github.io/leakcanary/) is a powerful tool for detecting memory leaks in Android applications. However, there are scenarios where you might want to disable it, such as when preparing a debug build for performance testing or when it's not needed.
+
+### Disabling LeakCanary via Gradle command
+
+You can disable LeakCanary manually by passing the `-PnoLeakCanary` flag in the Gradle command. For example:
+
+```bash
+./gradlew app:assembleFullDebug -PnoLeakCanary
+```
+
+This flag ensures that LeakCanary is excluded from the build.
+
+### Disabling LeakCanary via properties file
+
+Alternatively, you can disable LeakCanary by setting the noLeakCanary property in the gradle.properties file. This can be done at either the project level or the home level.
+
+```properties
+noLeakCanary=true
+```
+
+::::warning
+If it is set you need to update the lockfile otherwise gradle complains about the deps being not used.
+It can be done with
+
+```bash
+./gradlew alldependencies --write-locks
+```
+
+::::
+
+## Best practices for using LeakCanary
+
+- **Enable only in debug builds:** LeakCanary should only be included in debug builds to avoid impacting performance in production builds.
```

Comment:
Is this needed in the documentation? If someone were to submit a PR to change this we would reject it and the PR already hints at it being debug-only.